src/embeddings.rs

```diff
@@ -513,6 +513,29 @@ impl<'a> Iterator for IterWithNorms<'a> {
     }
 }
 
+/// Reconstructable embedding matrix
+pub trait Reconstruct<V> {
+    /// Reconstruct the embedding matrix.
+    ///
+    /// This method reconstruct an embedding storage and construct a new
+    /// `Embeddings` struct with storage type of `NdArray`.
+    fn reconstruct(&self) -> Embeddings<V, NdArray>;
+}
+
+impl<V> Reconstruct<V> for Embeddings<V, QuantizedArray>
+where
+    V: Vocab + Clone,
+{
+    fn reconstruct(&self) -> Embeddings<V, NdArray> {
```

Comment:
I'd prefer this to take `self` instead of `&self`, that way the user of the API can decide whether they want to keep the old embeddings or drop them. I'm not sure whether this trait could be replaced by implementing `From`, we might run into problems with not having specialization. If it's possible, we could then add another method on an impl for
~~~Rust
impl<V, S> Embeddings<V, S> where V: Vocab + Clone, S: Reconstruct {
    fn reconstruct(self) -> Embeddings<V, NdArray> {...}
}

impl<V, S> From<Embeddings<V,S> for Embeddings<V, NdArray> where S: Reconstruct {
    fn from(reconstructable: Embeddings<V, S>) -> Self {
        reconstructable.reconstruct()
    }
}
~~~